pkgs/tools/security/john/default.nix

```diff
@@ -25,6 +50,7 @@ stdenv.mkDerivation rec {
 
   preConfigure = ''
     cd src
+    python ./opencl_generate_dynamic_loader.py  # Update opencl_dynamic_loader.c
```

Comment:
this step should be handled with `optionalString`

Reply:
Done, but I should point out that running this script is a no-op when we haven't modified its source.

Reply:
In nixpkgs, people have been wanting to move away from `lib.optional`, because, if you accidentally used `[ example ]` with it, you end up with nested lists instead of them merging together.

Reply:
I see, thank you for the explanation. (I think you meant to post this as a reply on the other thread.)

Reply:
yup, lmao